Frontend/ui-driver/src/Flow.purs

```diff
@@ -3155,8 +3182,97 @@ referralFlow = do
       referralType = case (cityConfig.showDriverReferral || appConfig.enableDriverReferral), (cityConfig.showCustomerReferral || appConfig.enableCustomerReferral) of
                       false, true -> ST.CUSTOMER
                       _, _ -> ST.DRIVER
-  modifyScreenState $ DriverReferralScreenStateType (\driverReferralScreen -> driverReferralScreen { props {driverReferralType = referralType}})
-  driverReferralScreenFlow
+  modifyScreenState $ BenefitsScreenStateType (\benefitsScreen -> benefitsScreen { props {driverReferralType = referralType}})
+  benefitsScreenFlow
+
+lmsVideoScreenFlow :: FlowBT String Unit
+lmsVideoScreenFlow = do
+  modifyScreenState $ LmsVideoScreenStateType (\lmsVideoScreen -> lmsVideoScreen { props {showShimmer = true}})
+  lmsVideoScreenAction <- UI.lmsVideoScreen
+  case lmsVideoScreenAction of
+    GO_TO_QUIZ_SCREEN state-> do
+      res <- lift $ lift $ Remote.getAllLmsQuestions (state.props.selectedModule ^. _moduleId) $ HU.getLanguageTwoLettersForString state.props.selectedLanguage
+      case res of
+        Right resp -> do
+            let (API.LmsGetQuizRes extractedData ) = resp
+            modifyScreenState $ LmsQuizScreenStateType (\lmsQuizScreen -> lmsQuizScreen { data {questions = transformQuizRespToQuestions resp}, props {selectedLanguage = state.props.selectedLanguage, selectedTranslatedModule = extractedData.selectedModuleInfo, currentQuestionIndex = 0, currentQuestionSelectedOptionsData = {selectedSingleOption : Nothing, selectedMultipleOptions : []}}})
+        Left err -> do
+            pure $ toast $ getString UNABLE_TO_LOAD_QUIZ_PLEASE_TRY_AGAIN
+            lmsVideoScreenFlow
+      modifyScreenState $ LmsVideoScreenStateType (\lmsVideoScreen -> lmsVideoScreen { props {isFetchingQuiz = false}})
+      lmsQuizFlow
+    REFRESH_LMS_VIDEO_SCREEN state -> lmsVideoScreenFlow
+    GO_TO_BENEFITS_SCREEN -> do
+      modifyScreenState $ BenefitsScreenStateType (\benefitsScreen -> benefitsScreen { props {showShimmer = true}, data {moduleList = {completed : [], remaining : []}}})
+      benefitsScreenFlow
+    SELECT_LANGUAGE_FOR_VIDEOS state -> do
+      let genLanguageList = HU.generateLanguageList $ state.props.selectedModule ^. _languagesAvailableForVideos
+      modifyScreenState $ SelectLanguageScreenStateType (\selectLangState -> selectLangState{ data {languageList = genLanguageList}, props{selectedLanguage = state.props.selectedLanguage, onlyGetTheSelectedLanguage = true, selectLanguageForScreen = "LMS_VIDEO_SCREEN" }})
+      selectLanguageFlow
+      modifyScreenState $ SelectLanguageScreenStateType (\selectLangState -> selectLangState{ props{ onlyGetTheSelectedLanguage = false, selectedLanguage = "", selectLanguageForScreen = ""}})
+      lmsVideoScreenFlow
+
+lmsQuizFlow :: FlowBT String Unit
+lmsQuizFlow = do
```

Comment:
please write this flow function little cleaner -
- Divide some parts into subfunctions
- Don't have very lengthy lines of code in one line
- use proper indentation